artifact/gem5art/artifact/_artifactdb.py

```diff
@@ -191,19 +192,171 @@ def searchByLikeNameType(self, name: str, typ: str, limit: int) -> Iterable[Dict
         """Returns an iterable of all artifacts in the database that match
         some type and a regex name."""
 
-        data = self.artifacts.find({'type':typ,
+        data = self.artifacts.find({'type': typ,
                                     'name': {'$regex': '{}'.format(name)}
                                    },
                                    limit=limit)
         for d in data:
             yield d
 
+
+class ArtifactFileDB(ArtifactDB):
+    """
+    This is a file-based database where Artifacts (as defined in artifacts.py)
+    are stored in a JSON file. This database does not keep track of files.
+
+    This database stores the data in three collections:
+    - artifacts: This stores the json serialized Artifact class
```

Comment:
What are the other 2 collection (objects in json nomenclature)

Reply:
Fixed